crates/ruff_linter/src/rules/pylint/rules/useless_else_on_loop.rs

```diff
@@ -91,17 +132,50 @@ fn loop_exits_early(body: &[Stmt]) -> bool {
     })
 }
 
-/// PLW0120
-pub(crate) fn useless_else_on_loop(
-    checker: &mut Checker,
+/// Generate a [`Fix`] to remove the `else` clause from the given statement.
+fn remove_else(
```

Comment:
I moved the handling out to its own function so that it can return `Result` and make all the error cases explicit (rather than using `unwrap`, which panics on failure).